tests/PedantryTest.php

```diff
@@ -86,7 +87,8 @@ public function provideProjectClassNames()
                 continue;
             }
 
-            $classNames[][] = 'MongoDB\\' . str_replace(DIRECTORY_SEPARATOR, '\\', substr($file->getRealPath(), strlen($srcDir) + 1, -4));
+            $className = 'MongoDB\\' . str_replace(DIRECTORY_SEPARATOR, '\\', substr($file->getRealPath(), strlen($srcDir) + 1, -4));
+            $classNames[$className][] = $className;
```

Comment:
You could yield the result directly.
```suggestion
            yield $className => [$className];
```

Reply:
An evil code shortcut:
```suggestion
            yield $className => $className = 'MongoDB\\' . str_replace(DIRECTORY_SEPARATOR, '\\', substr($file->getRealPath(), strlen($srcDir) + 1, -4));;
```

https://3v4l.org/ntqfH

Reply:
I'd leave that out to not have more changes than necessary here. That said, I've created PHPLIB-1191 to track making data providers static and refactoring them to return `Generator` instances instead of arrays.